optimum/intel/openvino/modeling_timm.py

```diff
@@ -0,0 +1,338 @@
+from transformers import PretrainedConfig, PreTrainedModel
+from transformers.image_processing_utils import BaseImageProcessor, BatchFeature, get_size_dict
+from transformers.image_transforms import resize, to_channel_dimension_format
+from transformers.image_utils import (
+    IMAGENET_STANDARD_MEAN,
+    IMAGENET_STANDARD_STD,
+    ChannelDimension,
+    ImageFeatureExtractionMixin,
+    ImageInput,
+    PILImageResampling,
+    make_list_of_images,
+    to_numpy_array,
+    valid_images,
+)
+from transformers.modeling_outputs import ImageClassifierOutput
+from transformers.utils import TensorType
+
+from optimum.exporters.onnx.config import VisionOnnxConfig
+from optimum.utils import NormalizedVisionConfig
+
+
+set_fused_attn(False, False)
+
+
+def is_timm_ov_dir(model_dir):
+    config_file = None
+    has_xml = False
+    has_bin = False
+    if os.path.isdir(model_dir):
+        for filename in glob(os.path.join(model_dir, "*")):
+            if filename.endswith(".xml"):
+                has_xml = True
+            if filename.endswith(".bin"):
+                has_bin = True
+            if filename.endswith("config.json"):
+                config_file = filename
+    if config_file and has_xml and has_bin:
+        with open(config_file) as conf:
+            hf_hub_id = json.load(conf).get("hf_hub_id", None)
+        if hf_hub_id and model_info(hf_hub_id).library_name == "timm":
+            return True
+    return False
+
+
+class TimmConfig(PretrainedConfig):
+    model_type = "timm"
+
+    @classmethod
+    def from_pretrained(
+        cls,
+        pretrained_model_name_or_path: Union[str, os.PathLike],
+        cache_dir: Optional[Union[str, os.PathLike]] = None,
+        force_download: bool = False,
+        local_files_only: bool = False,
+        token: Optional[Union[str, bool]] = None,
+        revision: str = "main",
+        **kwargs,
+    ) -> "PretrainedConfig":
+        if is_timm_ov_dir(pretrained_model_name_or_path):
+            config_path = os.path.join(pretrained_model_name_or_path, "config.json")
+            return cls.from_json_file(config_path)
+
+        kwargs["cache_dir"] = cache_dir
+        kwargs["force_download"] = force_download
+        kwargs["local_files_only"] = local_files_only
+        kwargs["revision"] = revision
+
+        config_dict = load_model_config_from_hf(pretrained_model_name_or_path)[0]
+        config_dict["num_labels"] = config_dict.pop("num_classes")
+        config_dict["image_size"] = config_dict.get("input_size")[-1]
+
+        return cls.from_dict(config_dict, **kwargs)
+
+
+class TimmOnnxConfig(VisionOnnxConfig):
+    DEFAULT_TIMM_ONNX_OPSET = 13
+    outputs = OrderedDict([("logits", {0: "batch_size"})])
+    NORMALIZED_CONFIG_CLASS = NormalizedVisionConfig
+    MIN_TORCH_VERSION = version.parse("1.11")
+
+    @property
+    def inputs(self) -> Dict[str, Dict[int, str]]:
+        return {"pixel_values": {0: "batch_size", 1: "num_channels", 2: "height", 3: "width"}}
```

Comment:
Is this to support the ONNX export of timm models? Do you think it would be useful to move it (later) in optimum main repo?